testTypeIperf.py

```diff
@@ -64,8 +64,8 @@ def can_run_reverse(self) -> bool:
         return False
 
 
-test_type_handler_iperf_tcp = TestTypeHandlerIperf(TestType.IPERF_TCP)
-test_type_handler_iperf_udp = TestTypeHandlerIperf(TestType.IPERF_UDP)
+TestTypeHandler.register_test_type(TestTypeHandlerIperf(TestType.IPERF_TCP))
```

Comment:
If we do "register_test_type" as part of the main function prior to creating threads. Then we don't need the lock right?

I don't think the register functions need to be called per file. We can consolidate it all in one place as part of the start up of the scripts.

(This comment applies to plugins as well)

What is the purpose of registering? Are we expecting users to remove plugins?

Reply:
> What is the purpose of registering? Are we expecting users to remove plugins?

We have a `get()` method, that can lookup plugins/test-type-handlers by name/ID. It could be implemented in various ways. For example, previously `TestTypeHandler.get()` had an `if-elif-else` block and named every ID. This now changed to use a dictionary. I think that is slightly nicer (see below).

> If we do "register_test_type" as part of the main function prior to creating threads. Then we don't need the lock right?

We could require that the user calls a setup/initialize method first before being allowed to call `get()`. This pushes an additional requirement to the user (note that `main.py`, `evaluator.py` and unit tests would all need to follow this requirement). I think it is nicer that any caller can just call `get()` at any time (and on any thread), and it will initialize itself in a thread-safe way. One thing less a user can do wrong.

---

This PR moves some code around and changes who does what where. It's not a major change, but I think this is nicer because:

- it's fewer lines of code, which is often an indication that it's better
- if we add a new plugin/test-type, then we require only one single line outside of the plugin/test-type module (the `import newModule`). Additionally, it needs the `register()` call, but that's inside the new module itself.
- we create new plugin/test-type via copy+paste. It's pretty obvious how you must call the `register()` method (just copy and adjust). Also, there are unit tests, so it would be pretty hard to create a new plugin/test-type and make a mistake in how it registers itself.

No major difference, but slightly nicer IMO.

Reply:
I want that there is less work when a new plugin/test-types is added.

With this PR, a plugin/test-type requires one `import newModule` and one `register()` call to be hooked up. We could drop the `import newModule` if we would search the file system for `plugin*.py`/`testType*.py` files. But that seems worse, because we should be explicit which modules are loaded. So two lines is minimal.